www/src/components/MeasuredItem.jsx

```diff
@@ -1,43 +1,50 @@
-import React from 'react';
+import React, {
+  useContext,
+  useState,
+  useRef,
+  useEffect,
+} from 'react';
 import PropTypes from 'prop-types';
 
-export default class MeasuredItem extends React.Component {
-  constructor(props) {
-    super(props);
-
-    const initialMeasurements = props.properties.reduce((acc, property) => {
-      acc[property] = null;
-      return acc;
-    }, {});
-
-    this.state = initialMeasurements;
-    this.item = React.createRef();
-  }
-
-  componentDidMount() {
-    // Needs a moment to render children to DOM first.
-    setTimeout(this.measure.bind(this), 10);
-  }
-
-  measure() {
-    const computedStyle = getComputedStyle(this.item.current);
-    const measurements = this.props.properties.reduce((acc, property) => {
-      acc[property] = computedStyle.getPropertyValue(property);
-      return acc;
-    }, {});
-
-    this.setState(measurements);
-  }
-
-  render() {
-    return (
-      <>
-        {this.props.renderBefore ? this.props.renderBefore(this.state) : null}
-        {React.cloneElement(this.props.children, { ref: this.item })}
-        {this.props.renderAfter ? this.props.renderAfter(this.state) : null}
-      </>
-    );
-  }
+import { SettingsContext } from '../context/SettingsContext';
+
+const initialMeasurements = {};
+
+const MeasuredItem = ({
+  properties,
+  renderBefore,
+  renderAfter,
+  children,
+}) => {
+  const { theme } = useContext(SettingsContext);
+  const [measurements, setMeasurements] = useState(initialMeasurements);
+  const itemRef = useRef();
+
+  useEffect(
+    () => {
+      const measure = () => {
+        const computedStyle = getComputedStyle(itemRef.current);
+        const measurements = properties.reduce((acc, property) => {
+          acc[property] = computedStyle.getPropertyValue(property);
+          return acc;
+        }, {});
+        setMeasurements(measurements);
+      };
+      // Needs a moment to finish switching theme and re-render children to DOM first.
```

Comment:
Does a set timeout work on very slow browsers?

I wonder if you can work around using `setTimeout` by using another `useEffect` that handles calling `measure`. Using the cleanup feature of `useEffect` might also help? 

This feels like a super small edge case I don't feel very strongly about it.

Reply:
@manny-m If the new styles don't apply within a second, the `setTimeout` indeed will not work as expected. I tried different approaches like you mentioned, using `onload`/`onreadystatechange`, react-helmet's `onChangeClientState`... all without luck unfortunately lol.

I'm going to say the `setTimeout` will suffice for now but will file a ticket to see if we can refactor to avoid the `setTimeout` in the Paragon backlog.